packages/lean_analytics/lib/core/event.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:flutter/foundation.dart';
+
+class AnalyticsEvent {
+  AnalyticsEvent({
+    required this.name,
+    this.params,
+  });
+
+  final Map<String, Object>? params;
```

Comment:
Do we want to differentiate the null and empty case?

Reply:
Semantically it doesn't make any difference, so I wouldn't bother to be honest

Reply:
Then its probably easier if we dont use nullables anywhere.